bolt-cli/src/commands/send.rs

```diff
@@ -0,0 +1,159 @@
+use alloy::{
+    eips::eip2718::Encodable2718,
+    network::{EthereumWallet, TransactionBuilder},
+    primitives::{keccak256, B256, U256},
+    providers::{ProviderBuilder, SendableTx},
+    rpc::types::TransactionRequest,
+    signers::{local::PrivateKeySigner, Signer},
+};
+use eyre::{bail, Context, Result};
+use rand::Rng;
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tracing::info;
+
+use crate::cli::SendCommand;
+
+/// Path to the lookahead endpoint on the Bolt RPC server.
+const BOLT_LOOKAHEAD_PATH: &str = "proposers/lookahead";
+
+impl SendCommand {
+    /// Run the `send` command.
+    pub async fn run(self) -> Result<()> {
+        let wallet: PrivateKeySigner = self.private_key.parse().wrap_err("invalid private key")?;
+        let transaction_signer = EthereumWallet::from(wallet.clone());
+
+        let provider = ProviderBuilder::new()
+            .with_recommended_fillers()
+            .wallet(transaction_signer)
+            .on_http(self.bolt_rpc_url.clone());
+
+        // Fetch the lookahead info from the Bolt RPC server
+        let mut lookahead_url = self.bolt_rpc_url.join(BOLT_LOOKAHEAD_PATH)?;
+
+        // Note: it's possible for users to override the target sidecar URL
+        // for testing and development purposes. In most cases, the sidecar will
+        // reject a request for a slot that it is not responsible for.
+        let target_url = if let Some(sidecar_url) = self.override_bolt_sidecar_url {
+            // If using the override URL, we don't need to fetch the active proposers only.
+            // we will set the next slot as the target slot.
+            sidecar_url.clone()
+        } else {
+            // Filter out slots that are not active or in the past, to fetch the next
+            // active proposer slot.
+            lookahead_url.set_query(Some("activeOnly=true&futureOnly=true"));
+            self.bolt_rpc_url.clone()
+        };
+
+        let lookahead_res = reqwest::get(lookahead_url).await?.json::<Vec<LookaheadSlot>>().await?;
+        if lookahead_res.is_empty() {
+            println!("no bolt proposer found in the lookahead, try again later 🥲");
+            return Ok(());
+        }
+
+        // Extract the next preconfirmer slot from the lookahead info
+        let target_slot = lookahead_res[0].slot;
```

Comment:
Can we do `.get(0).expect("...")` instead?

Reply:
we check if it's empty in the line above, so that would be unreachable